### PR TITLE
update README, include serverless-offline and instructions for use

### PR DESCRIPTION
**Pull request by philvarner**
Include directions for running locally using serveless-offline, and other updates to the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,75 +2,111 @@
...
+| Name                                                       | STAC Version | STAC API Version | Description                         |
+| ---------------------------------------------------------- | ------------ | ---------------- |
+| [Earth Search](https://earth-search.aws.element84.com/v0/) | 1.0.0-beta.2 | 0.9.0            | Catalog of some AWS Public Datasets |
+| [Landsat Look](https://landsatlook.usgs.gov/stac-server)   | 1.0.0        | 0.9.0            |                                     |
+
+## Migration
+
+### 0.3 -> 0.4
 
+The Serverless Framework supports provisioning AWS resources, but it does not support updating existing resources. In 0.4, the default Elasticsearch version has been updated from 7.9 to 7.10. Continuing to use 7.9 should not cause any problems, but it recommended that you manually upgrade to 7.10 by going to [AWS Console - Amazon OpenSearch Service](https://console.aws.amazon.com/esv3/home), choosing the Elasticsearch domain used by your stac-server deployment (e.g., stac-server-{stage}-es), choose Upgrade from the Actions menu, and then upgrade to Elasticsearch 7.10.
```
Comment (philvarner) [inline README.md:41]: the actual update for 7.10 is going to be in another PR, but I was working on it at the same time, and put it in the README along with all of these other changes.

File: README.md [hunk 2]
```diff
@@ -2,75 +2,111 @@
...
-| STAC_DOCS_URL | URL to documentation | [https://stac-utils.github.io/stac-server](https://stac-utils.github.io/stac-server) |
-| ES_BATCH_SIZE | Number of records to ingest in single batch | 500 |
-| LOG_LEVEL | Level for logging (CRITICAL, ERROR, WARNING, INFO, DEBUG) | INFO |
-| STAC_API_URL | The root endpoint of this API | Inferred from request |
-| ENABLE_TRANSACTIONS_EXTENSION | Boolean specifying if the [Transaction Extension](https://github.com/radiantearth/stac-api-spec/tree/master/extensions/transaction) should be activated | false |
-
-After reviewing the settings, build and deploy the project.
+Copy the [example serverless config file](serverless.yml.example) to a file named `serverless.yml`:
 
+```sh
```
Comment (philvarner) [inline README.md:62]: naming the file serverless.yml.example and requring users to copy it makes it less likely that a dev will accidentally commit local changes to their serverless file, and that a merge from the upstream will overwrite customer changes.

File: README.md [hunk 3]
```diff
@@ -2,75 +2,111 @@
...
 ```
-$ npm install
-$ npm run build
-$ npm run deploy
+
+There are some settings that should be reviewed and updated as needeed in the serverless config file, under provider->environment:
+
+| Name                          | Description                                                                                                                                             | Default Value                                                                        |
+| ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------ |
+| STAC_VERSION                  | STAC Version of this STAC API                                                                                                                           | 1.0.0                                                                                |
```
Comment (philvarner) [inline README.md:70]: I removed STAC_API_VERSION here, since it no longer exists

File: README.md [hunk 4]
```diff
@@ -93,21 +129,82 @@ Errors that occur during ingest will end up in the dead letter processing queue,
...
 # Install dependencies in package.json
-$ npm install
+npm install
 
 # Run the build command in each of the packages (runs webpack)
-$ npm run build
+npm run build
+
+# Run ESLint
+npm run lint
```
Comment (philvarner) [inline README.md:152]: this command is in a different commit, but will eventually exist in the codebase

File: README.md [hunk 5]
```diff
@@ -93,21 +129,82 @@ Errors that occur during ingest will end up in the dead letter processing queue,
...
+also set the STAC_API_URL explicitly:
+
+```
+STAC_API_URL: http://localhost:3000
+```
+
+Then, use npm `serve` command to run serverless offline:
+
+```
+npm run serve
```
Comment (philvarner) [inline README.md:187]: serve uses serverless-offline

File: serverless.yml.example [hunk 6]
```diff
@@ -30,7 +30,7 @@ provider:
           - sqs:DeleteMessage
         Resource:
           Fn::GetAtt: [ingestQueue, Arn]
-
+  lambdaHashingVersion: 20201221
```
Comment (philvarner) [inline serverless.yml.example:33]: this addition is a recommendation from the serverless deploy command

**Review by matthewhanson — APPROVED** [review]
